src/test/test_bitcoin_fuzzy.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2009-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#include <stdint.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include <pubkey.h>
+#include <streams.h>
+#include <util/memory.h>
+
+const std::function<std::string(const char*)> G_TRANSLATION_FUN = nullptr;
+
+static bool read_stdin(std::vector<uint8_t>& data)
+{
+    uint8_t buffer[1024];
+    ssize_t length = 0;
+    while ((length = read(STDIN_FILENO, buffer, 1024)) > 0) {
+        data.insert(data.end(), buffer, buffer + length);
+
+        if (data.size() > (1 << 20)) return false;
+    }
+    return length == 0;
+}
+
+static void test_one_input(std::vector<uint8_t> buffer);
+
+static std::unique_ptr<ECCVerifyHandle> globalVerifyHandle;
+void initialize()
+{
+    globalVerifyHandle = MakeUnique<ECCVerifyHandle>();
+}
+
+// This function is used by libFuzzer
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size)
+{
+    test_one_input(std::vector<uint8_t>(data, data + size));
+    return 0;
+}
+
+// This function is used by libFuzzer
+extern "C" int LLVMFuzzerInitialize(int* argc, char*** argv)
```

Comment:
Nit: Could drop unused variable names `argc` and `argv`.

Reply:
Related: https://llvm.org/docs/LibFuzzer.html#startup-initialization